src/main/java/dev/waterdog/utils/Configuration.java

```diff
@@ -38,10 +51,17 @@ public Configuration(Path path) {
     public Configuration(File file) {
         this.file = file;
 
-        if (!this.file.exists()) {
+        if (this.file != null && !this.file.exists()) {
             try {
-                this.file.getParentFile().mkdirs();
-                this.file.createNewFile();
+                File parentFile = this.file.getParentFile();
+
+                if (parentFile != null) {
+                    parentFile.mkdirs();
+                }
+
+                FileWriter myWriter = new FileWriter(file);
```

Comment:
Why is empty config written to file here? There is no point to save empty config, should be same as empty file.

Reply:
Your previous approach of saving the file used to create an empty file 
![image](https://user-images.githubusercontent.com/27902328/109652810-3bfbc580-7b2e-11eb-8a72-b43bcf19d1e6.png) and that's why I created this method. Becuse when you try to json.parse or yaml.parse from an empty file it will give you an error :)